manifests/database/default_read_grant.pp

```diff
@@ -2,10 +2,10 @@
 #
 # @api private
 define puppetdb::database::default_read_grant (
-  String $database_name,
-  String $schema,
-  String $database_username,
-  String $database_read_only_username,
+  String[1]              $database_name,
+  String                 $schema,
```

Comment:
String without size is questionable.

Prefer this if an empty string is not valid:
```suggestion
  String[1]              $schema,
```

Or this if an empty string is valid:
```suggestion
  String[0]              $schema,
```

Without an explicit size, we don't know if the lower bound is 0 chars or just not specified.

Reply:
Thanks @smortex for the suggestion, I've set lower bounds for both strings and integers in the latest commit. I've checked each one for usage of empty strings '' and to guard against negative integers.